### PR TITLE
Add support for the Skagen Falster 2.

### PR DESCRIPTION
**Pull request by MagneFire**
This build is very similar the the `skipjack` and `firefish`(will need to work on this one) build.

It is based on `sawfish` where some patches were applied on top of `sawfish` to support the latest Pie preview (a bit bugged on `sawfish` but much improved performance.)

This watch uses a generic hybris tar in combination with a system tarball that is tuned for `ray` but based on the Pie preview from `sawfish`. This generic hybris tar has the major advantage in that it allows for adding support for new Oreo based watches without needing the recompile the entire tarball, either a system tarball can be provided or the system partition can be mounted. (Small note about Pie preview, it is compatible with Oreo, the actual Pie release isn't.)

### CODE REVIEW

File: conf/machine/ray.conf [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+#@TYPE: Machine
+#@NAME: Huawei Watch
+#@DESCRIPTION: Machine configuration for the Huawei Watch
```
Comment (jrtberlin) [inline conf/machine/ray.conf]: You might want to change the values here ;)

File: recipes-android/brcm-patchram-plus/brcm-patchram-plus_git.bbappend [hunk 2]
```diff
@@ -1,3 +1,3 @@
-FILESEXTRAPATHS_prepend_sawfish := "${THISDIR}/brcm-patchram-plus:"
-SRC_URI_append_sawfish = " file://patchram.service "
-CFLAGS_append_sawfish = " -DLPM_SAWSHARK"
+FILESEXTRAPATHS:prepend:ray := "${THISDIR}/brcm-patchram-plus:"
+SRC_URI:append:ray = " file://patchram.service "
+CFLAGS:append:ray = " -DLPM_SAWSHARK"
```
Comment (jrtberlin) [inline recipes-android/brcm-patchram-plus/brcm-patchram-plus_git.bbappend]: are you sure about `-DLPM_SAWSHARK`?

File: recipes-kernel/linux/linux-ray_n.bb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+require recipes-kernel/linux/linux.inc
+inherit gettext
+
+SECTION = "kernel"
+SUMMARY = "Android kernel for the Huawei Watch 2 Bluetooth"
```
Comment (jrtberlin) [inline recipes-kernel/linux/linux-ray_n.bb]: Still not the Huawei Watch 2 :D

**Review by jrtberlin — COMMENTED** [review]
Just some minor strings that are not changed yet.

File: conf/machine/ray.conf (hunk 1 above)
Comment (MagneFire) [reply conf/machine/ray.conf]: Great catch ;P

File: recipes-android/brcm-patchram-plus/brcm-patchram-plus_git.bbappend (hunk 2 above)
Comment (MagneFire) [reply recipes-android/brcm-patchram-plus/brcm-patchram-plus_git.bbappend]: Bluetooth works differently on the Skagen Falster 2 (uses hci-smd, similar to `swift`).
It's not functional for the moment anyway.
Will remove the recipe for now.

**Review by jrtberlin — APPROVED** [review]

**Comment by jrtberlin** [comment]
Thanks!

**Pull request merged**